src/domain-services/flows/flow-search-service.ts

```diff
@@ -0,0 +1,652 @@
+      const reportDetails = reportDetailsMap.get(flow.id) ?? [];
+
+      let reportDetailsWithChannel: ReportDetail[] = [];
+      if (reportDetails.length > 0) {
+        reportDetailsWithChannel =
+          await this.categoryService.addChannelToReportDetails(
+            models,
+            reportDetails
+          );
+      }
+
+      let parkedParentSource: FlowParkedParentSource | null = null;
+      const shouldLookAfterParentSource =
+        flowLink.length > 0 && shouldIncludeChildrenOfParkedFlows;
+
+      if (shouldLookAfterParentSource) {
+        parkedParentSource = await this.flowService.getParketParents(
+          flow,
+          flowLink,
+          models
+        );
+      }
+
+      const childIDs: number[] =
+        flowLinksMap
+          .get(flow.id)
+          ?.filter(
+            (flowLink) => flowLink.parentID === flow.id && flowLink.depth > 0
+          )
+          .map((flowLink) => flowLink.childID.valueOf()) ?? [];
+
+      const parentIDs: number[] =
+        flowLinksMap
+          .get(flow.id)
+          ?.filter(
+            (flowLink) => flowLink.childID === flow.id && flowLink.depth > 0
+          )
+          .map((flowLink) => flowLink.parentID.valueOf()) ?? [];
+
+      const parsedFlow: Flow = this.buildFlowDTO(
+        flow,
+        categoriesByVersion,
+        organizations,
+        locations,
+        plans,
+        usageYears,
+        childIDs,
+        parentIDs,
+        externalReferences,
+        reportDetailsWithChannel,
+        parkedParentSource
+      );
+
+      return parsedFlow;
+    });
+    const items = await Promise.all(promises);
+
+    return {
+      flows: items,
+      hasNextPage: hasNextPage,
+      hasPreviousPage: nextPageCursor !== undefined,
+      prevPageCursor: nextPageCursor ? nextPageCursor - limit : 0,
+      nextPageCursor: hasNextPage
+        ? nextPageCursor
+          ? nextPageCursor + limit
+          : limit
+        : 0,
+      pageSize: flows.length,
+      sortField: `${orderBy.entity}.${orderBy.column}` as FlowSortField,
+      sortOrder: sortOrder ?? 'desc',
+      total: count,
+    };
+  }
+
+  /**
+   * This method returns the shortcut filter defined with the operation
+   * IN if is true or NOT IN if is false
+   *
+   * @param isPendingFlows
+   * @param isCommitmentFlows
+   * @param isPaidFlows
+   * @param isPledgedFlows
+   * @param isCarryoverFlows
+   * @param isParkedFlows
+   * @param isPassThroughFlows
+   * @param isStandardFlows
+   * @returns [{ category: String, operation: Op.IN | Op.NOT_IN}]
+   */
+  mapShortcutFilters(
+    isPendingFlows: boolean,
+    isCommitmentFlows: boolean,
+    isPaidFlows: boolean,
+    isPledgedFlows: boolean,
+    isCarryoverFlows: boolean,
+    isParkedFlows: boolean,
+    isPassThroughFlows: boolean,
+    isStandardFlows: boolean
+  ): ShortcutCategoryFilter[] | null {
+    const filters = [
+      { flag: isPendingFlows, category: 'Pending', id: 45 },
```

Comment:
I don't think it is good idea to hardcode `id`.

Reply:
It was a workaround in order to reduce the amount of queries to perform and reduce the execution time but now it makes no sense.
I've updated it!